faculty-bios/php/faculty-bio-listing.php

```diff
@@ -24,18 +24,23 @@ function create_faculty_bio_listing($schools, $cas, $caps, $gs, $sem, $displayFa
     $found_lead = false;
     $found_faculty = false;
     foreach( $bios as $bio){
```

Comment:
Looking at the structure of this for loop, it seems that it's iterating through all objects until it's either found both the lead and the faculty, at which point it keeps iterating until it has no more bios. Perhaps an enhancement at the end of the for loop could be something like:
```
if( $found_lead && $found_faculty) {
    break;
}
```
So that it doesn't have to iterate through all bios in the array if it has already found both values.

Reply:
There are 2 parts to this. Part 1 is to display the headers. Part 2 is line 44: print the html for each individual bio. I'm guessing you just didn't see line 44?

Reply:
i just adda couple of comments to help clarify

Reply:
Oh. I read that line as outside the for loop, sorry.

Reply:
Yeah, this version of PHP is the worst. We have to do SOOO many extra checks for everything.